packages/common/src/dbt/validation.ts

```diff
@@ -0,0 +1,158 @@
+} from '../types/dbt';
+import { ParseError, UnexpectedServerError } from '../types/errors';
+import assertUnreachable from '../utils/assertUnreachable';
+import lightdashMetadataSchema from './schemas/lightdashMetadata.json';
+import dbtManifestSchemaV8 from './schemas/manifestV8.json';
+import dbtManifestSchemaV9 from './schemas/manifestV9.json';
+
+const LightdashMetadataSchemaRef =
+    'https://schemas.lightdash.com/dbt/manifest.json';
+
+const getDbtManifestWithLightdashMetadata = (dbtSchemaRef: string) => ({
+    definitions: {
+        LightdashNodeConfig: {
+            allOf: [
+                {
+                    $ref: `${dbtSchemaRef}#/definitions/NodeConfig`,
+                },
+                {
+                    type: 'object',
+                    properties: {
+                        meta: {
+                            $ref: '#/definitions/LightdashModelMetadata',
+                        },
+                    },
+                },
+            ],
+        },
+        LightdashColumnInfo: {
+            allOf: [
+                {
+                    $ref: `${dbtSchemaRef}#/definitions/ColumnInfo`,
+                },
+                {
+                    type: 'object',
+                    properties: {
+                        meta: {
+                            $ref: '#/definitions/LightdashColumnMetadata',
+                        },
+                    },
+                },
+            ],
+        },
+        LightdashCompiledModelNode: {
+            allOf: [
+                {
+                    $ref: `${dbtSchemaRef}#/definitions/ModelNode`,
+                },
+                {
+                    type: 'object',
+                    properties: {
+                        meta: {
+                            $ref: '#/definitions/LightdashModelMetadata',
+                        },
+                        config: {
+                            $ref: '#/definitions/LightdashNodeConfig',
+                        },
+                    },
+                },
+            ],
+        },
+    },
+    $schema: 'http://json-schema.org/draft-07/schema#',
+    $id: LightdashMetadataSchemaRef,
+});
+
+const getValidatorDefinition = (manifestVersion: DbtManifestVersion) => {
+    let dbtManifestSchema;
+    switch (manifestVersion) {
+        case DbtManifestVersion.V8:
+            dbtManifestSchema = dbtManifestSchemaV8;
+            break;
+        case DbtManifestVersion.V9:
+            dbtManifestSchema = dbtManifestSchemaV9;
+            break;
+        default:
+            return assertUnreachable(
+                manifestVersion,
+                new UnexpectedServerError(
+                    `Missing dbt manifest version "${manifestVersion}" in validation.`,
+                ),
+            );
+    }
+
+    const ajv = new Ajv({
+        schemas: [
+            getDbtManifestWithLightdashMetadata(dbtManifestSchema.$id),
+            lightdashMetadataSchema,
+            dbtManifestSchema,
```

Comment:
Our validation now depends on 3 schemas:
- **dbtManifestSchema**: is the original schema from dbt. The schema file depends on what version of dbt it was used to generate the manifest.
- **lightdashMetadataSchema**: contains the definitions for the Lightdash meta properties
- **getDbtManifestWithLightdashMetadata()**: combines the original dbt schema with Lightdash definitions